File: app/src/main/java/com/jworks/eigolens/ui/capture/ScopeLevel.kt
package com.jworks.eigolens.ui.capture

sealed class ScopeLevel {
    data class Word(val word: String) : ScopeLevel()
    data class Phrase(val text: String, val words: List<String>) : ScopeLevel()
    data class Sentence(val text: String) : ScopeLevel()
    data class Paragraph(val text: String) : ScopeLevel()
    data object FullSnapshot : ScopeLevel()
}

File: ipad-app/EigoLens/Domain/AI/ScopeLevel.swift
import Foundation

enum ScopeLevel: Equatable {
    case word(String)
    case phrase(text: String, words: [String])
    case sentence(String)
    case paragraph(String)
    case fullSnapshot

    var displayName: String {
        switch self {
        case .word: return "Word"
        case .phrase: return "Phrase"
        case .sentence: return "Sentence"
        case .paragraph: return "Paragraph"
        case .fullSnapshot: return "Full Text"
        }
    }

    var scopeName: String {
        switch self {
        case .word: return "word"
        case .phrase: return "phrase"
        case .sentence: return "sentence"
        case .paragraph: return "paragraph"
        case .fullSnapshot: return "full"
        }
    }
}
